src/main/kotlin/com/jetbrains/snakecharm/lang/psi/SmkSLSectionReferencesCollector.kt

```diff
@@ -0,0 +1,81 @@
+package com.jetbrains.snakecharm.lang.psi
+
+import com.intellij.lang.injection.InjectedLanguageManager
+import com.intellij.psi.PsiElement
+import com.intellij.psi.util.PsiTreeUtil
+import com.intellij.psi.util.elementType
+import com.intellij.psi.util.parentOfType
+import com.jetbrains.python.PyElementTypes
+import com.jetbrains.python.psi.PyElementVisitor
+import com.jetbrains.python.psi.PyKeywordArgument
+import com.jetbrains.python.psi.PyRecursiveElementVisitor
+import com.jetbrains.python.psi.PyStringLiteralExpression
+import com.jetbrains.snakecharm.lang.psi.elementTypes.SmkElementTypes
+import com.jetbrains.snakecharm.stringLanguage.lang.psi.SmkSLFile
+import com.jetbrains.snakecharm.stringLanguage.lang.psi.SmkSLReferenceExpressionImpl
+import com.jetbrains.snakecharm.stringLanguage.lang.psi.SmkSLSubscriptionExpressionImpl
+
+class SmkSLSectionReferencesCollector(
+    private val searchingType: String,
+    private val isRuleOrCheckpointAppropriate: (SmkRuleOrCheckpoint) -> Boolean = { true }
```

Comment:
For `use` syntax exection section will be missing, so we will likely got a FP on things like
```shell
rule foo:
   shell: "echo {log}"  # inspection warning here could be ignored, it is already reported, no error on runime, undefined sections are empty listrs

use rule foo as boo:
   log: "aa.log"  # <- likely FP here
```
so please add an issue for `use` support to cover uch case in this inspection.